src/main/java/org/elasticsearch/index/mapper/core/NumberFieldMapper.java

```diff
@@ -306,18 +306,14 @@ public final Filter termFilter(Object value, @Nullable QueryParseContext context
     public Filter fieldDataTermsFilter(List values, @Nullable QueryParseContext context) {
         IndexNumericFieldData fieldData = context.getForField(this);
         if (fieldData.getNumericType().isFloatingPoint()) {
-            // create with initial size large enough to avoid rehashing
-            DoubleOpenHashSet terms =
-                    new DoubleOpenHashSet((int) (values.size() * (1 + DoubleOpenHashSet.DEFAULT_LOAD_FACTOR)));
+            DoubleHashSet terms = new DoubleHashSet(values.size());
```

Comment:
DoubleHashSet is esoteric; should be a regular LongHashSet with fp normalization (Double.doubleBitsToLong).